runtime/src/message_processor.rs

```diff
@@ -75,8 +83,11 @@ fn verify_instruction(
 pub type ProcessInstruction =
     fn(&Pubkey, &mut [KeyedAccount], &[u8], u64) -> Result<(), InstructionError>;
 
+pub type SymbolCache = RwLock<HashMap<Vec<u8>, Symbol<instruction_processor_utils::Entrypoint>>>;
+
 pub struct MessageProcessor {
     instruction_processors: Vec<(Pubkey, ProcessInstruction)>,
+    symbol_cache: SymbolCache,
```

Comment:
Conflicts between the same name and different programs?

Reply:
wouldn't that be a problem with the non-cache path?

Reply:
This assumes the names are unique. Maybe thats not a good assumption.

Reply:
non-cache path its looked up, but with cached path, it might find another program's ptr.

Reply:
The lookup just seems to be a path based on the name and the fixed symbol name `instruction_processor_utils::ENTRYPOINT` which is the same for all programs. What is different in the Library::open path for each program other than the name?

Reply:
Its probably ok for now since the library is never unloaded but in the future if the library is closed we will have to clear the cache.